### PR TITLE
Working 404 Machine

### PR DESCRIPTION
**Pull request by Finanzcreeper**
We now have a working webserver that returns 404 to everything, valgring shows neither errors nor leaks!

### CODE REVIEW

File: interpreters/httpInterpreter.cpp [hunk 1]
```diff
@@ -0,0 +1,110 @@
+#include "httpInterpreter.hpp"
+
+void InterpretRequest(Request& request, const t_server& settings) {
```
Comment (hoeftint) [inline interpreters/httpInterpreter.cpp]: maybe its better readible when the function name is starting with a small letter

File: interpreters/httpInterpreter.cpp [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	headerField = request.HeaderFields.find("content-type");
+	if(headerField == request.HeaderFields.end()) {
+		return;
+	}
+
+	std::vector<std::string> allowedContentTypes;
+	std::vector<std::string> allowedContentSubtypes;
+
+	allowedContentTypes.push_back("hey/");
+	allowedContentSubtypes.push_back("ho");
```
Comment (hoeftint) [inline interpreters/httpInterpreter.cpp:77]: We could also add the allowed content types to the server struct later, so its not done for every request individually

File: interpreters/httpInterpreter.cpp [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+	while (allowedContentTypeIterator != allowedContentTypes.end()) {
+		if  (*allowedContentTypeIterator == foundContentType) {
+			contentTypeFound = true;
+			break;
+		}
+		++allowedContentTypeIterator;
+	}
+	if  (contentTypeFound == false) {
+		request.RequestIntegrity = UNSUPPORTED_MEDIA_TYPE;
+	} else {
```
Comment (hoeftint) [inline interpreters/httpInterpreter.cpp:97]:
```suggestion
	if  (contentTypeFound == ture) {
```

Comment (hoeftint) [reply interpreters/httpInterpreter.cpp:97]: The current if statement is in the code twice

**Review by hoeftint — CHANGES_REQUESTED** [review]
Nice code, just minor suggestions for better readibility

File: interpreters/httpInterpreter.cpp (hunk 3 above)
Comment (Finanzcreeper) [reply interpreters/httpInterpreter.cpp:97]: those are different is statements

File: interpreters/httpInterpreter.cpp (hunk 2 above)
Comment (Finanzcreeper) [reply interpreters/httpInterpreter.cpp:77]: could be done, but i dont know if it should be done. lets see in the future how to handle

File: interpreters/httpInterpreter.cpp (hunk 1 above)
Comment (Finanzcreeper) [reply interpreters/httpInterpreter.cpp]: good point

**Review by hoeftint — APPROVED** [review]

**Pull request merged**